web/src/lib/components/pages/record.tsx

```diff
@@ -364,6 +366,10 @@ export default class RecordPage extends Component<RecordProps, RecordState> {
     }
   }
 
+  areSentencesLoaded() {
+    return this.sentenceCache.length >= SET_COUNT;
+  }
```

Comment:
for derived properties like these, [typescript property accessors](https://www.typescriptlang.org/docs/handbook/classes.html#accessors) are nice IMHO.

```typescript
private get areSentencesLoaded() {
  return this.sentenceCache.length >= SET_COUNT;
}
```

usage:
```typescript
  while (!this.areSentencesLoaded) {
    ...
  }
```

To be clear: this has no advantage in terms of performance, security, type safety, or the likes. It's just a syntactical variant, which presents the derived property like a field when accessing it.
Therefore, using an accessor vs. a method here is mostly a matter of taste :)